### PR TITLE
oauth proxy support was never really implemented

### PR DESCRIPTION
**Pull request by josefkarasek**
https://jira.coreos.com/browse/LOG-201

@josefkarasek please backref the task associated with this PR

### CODE REVIEW

**Comment by josefkarasek** [comment]
`ElasticsearchSecure` now just says whether communication happens over tls or not. Should we use tls only now?

**Comment by ewolinetz** [comment]
Correct, we have no reason not to. Right @jcantrill @richm ?

File: pkg/apis/elasticsearch/v1alpha1/types.go [hunk 1]
```diff
@@ -85,11 +85,8 @@ type ElasticsearchNodeSpec struct {
 }
 
 // ElasticsearchSecure struct represents security configuration of the cluster
-// whether SearchGuard is enabled along with oauth-proxy sidecar
 type ElasticsearchSecure struct {
-	Disabled           bool   `json:"disabled"`
-	Image              string `json:"image,omitempty"`
-	CertificatesSecret string `json:"certificatesSecret,omitempty"`
+	CertificatesSecret string `json:"certificatesSecret"`
```
Comment (ewolinetz) [inline pkg/apis/elasticsearch/v1alpha1/types.go]: I dont think it necessary to keep this nested under `ElasticsearchSecure` if its the only element here.

**Comment by josefkarasek** [comment]
@ewolinetz let's merge, thx

**Pull request merged**